great_expectations/datasource/fluent/interfaces.py

```diff
@@ -517,10 +517,16 @@ def delete_asset(self, asset_name: str) -> None:
         Args:
             asset_name: name of DataAsset to be deleted.
         """
+        from great_expectations.data_context import CloudDataContext
+
         asset: _DataAssetT
+        asset = self.get_asset(asset_name=asset_name)
         self.assets = list(filter(lambda asset: asset.name != asset_name, self.assets))
 
-        self._save_context_project_config()
+        if self._data_context and isinstance(self._data_context, CloudDataContext):
+            self._data_context._delete_asset(id=str(asset.id))
+        else:
+            self._save_context_project_config()
```

Comment:
Hmm thinking out loud but is there a way we can avoid `isinstance` for a Cloud-specific path? Probably not

Reply:
I think there's an attribute we could check for, but that's essentially the same thing.

Reply:
I thought there was a property too but didn't find one. @cdkini - can you remind me why we should avoid isinstance where possible?